packages/block-editor/src/hooks/use-zoom-out.js

```diff
@@ -36,8 +42,14 @@ export function useZoomOut( zoomOut = true ) {
 	useEffect( () => {
 		if ( zoomOut ) {
 			setZoomLevel( 50 );
-		} else {
+		}
+		// Only reset zoom level if we were not in zoom out on mount:
+		// - Enter Zoom Level 50
+		// - Open inserter
+		// - Switch to Blocks Tab
+		// - Zoom level 50 should be persisted
+		else if ( ! zoomOut && ! isZoomOutOnMountRef.current ) {
```

Comment:
You're telling it you don't need to zoom out rather than turn off zoom out. So it's more a request of:
- false = "we don't need zoom out, but it's allowed"
- true = "we need to zoom out"

Reply:
Ok, that was not very clear. We might want to see if there are better alternatives to this hook later.